examples/platform/silabs/efr32/rs911x/hal/efx32_ncp_host.c

```diff
@@ -0,0 +1,506 @@
+
+#include "sl_wifi_constants.h"
+#include "sl_si91x_host_interface.h"
+#include "sl_board_configuration.h"
+#include "sl_status.h"
+#include "os_tick.h"   // CMSIS RTOS2
+#include "cmsis_os2.h" // CMSIS RTOS2
+#include "em_gpio.h"
+#include "em_cmu.h"
+#include "gpiointerrupt.h"
+#include "sl_si91x_status.h"
+#include "sl_constants.h"
+#include <stdbool.h>
+#include <string.h>
+
+#define VERIFY_STATUS(s)   \
+  do {                     \
+    if (s != SL_STATUS_OK) \
+      return s;            \
+  } while (0);
+
+#ifndef SL_WIFI_BOARD_READY_WAIT_TIME
+#define SL_WIFI_BOARD_READY_WAIT_TIME \
+  40000 //some scenarios like after firmware upgrade, it will take 40 seconds to boad ready
+#endif
+
+typedef struct {
+  sl_wifi_buffer_t *head;
+  sl_wifi_buffer_t *tail;
+  osMutexId_t mutex;
+  uint32_t flag;
+} si91x_packet_queue_t;
+
+osEventFlagsId_t si91x_events   = 0;
+osEventFlagsId_t si91x_bus_events   = 0;
+osEventFlagsId_t si91x_async_events = 0;
+osThreadId_t si91x_thread       = 0;
+osMutexId_t si91x_bus_mutex     = 0;
+osThreadId_t si91x_event_thread = 0;
+
+static si91x_packet_queue_t cmd_queues[SI91X_QUEUE_MAX];
+
+extern void si91x_bus_thread(void *args);
+extern void si91x_event_handler_thread(void *args);
+static void gpio_interrupt(uint8_t interrupt_number);
+sl_status_t sli_verify_device_boot(uint32_t *rom_version);
+sl_status_t sli_wifi_select_option(const uint8_t configuration);
+
+void sl_si91x_host_set_sleep_indicator(void)
+{
+  GPIO_PinOutSet(WFX_SLEEP_CONFIRM_PIN.port, WFX_SLEEP_CONFIRM_PIN.pin);
+}
+
+void sl_si91x_host_clear_sleep_indicator(void)
+{
+  GPIO_PinOutClear(WFX_SLEEP_CONFIRM_PIN.port, WFX_SLEEP_CONFIRM_PIN.pin);
+}
+
+uint32_t sl_si91x_host_get_wake_indicator(void)
+{
+  return GPIO_PinInGet(WAKE_INDICATOR_PIN.port, WAKE_INDICATOR_PIN.pin);
+}
+
+sl_status_t sl_si91x_host_init(void)

```

Comment:
Recommended : Add function header before each function.
giving brief what the function do